arbnode/batch_poster.go

```diff
@@ -1590,13 +1593,20 @@ func (b *BatchPoster) MaybePostSequencerBatch(ctx context.Context) (bool, error)
 		b.building.msgCount++
 	}
 
-	firstUsefulMsgTime := time.Now()
+	firstIncludedMsgTime := time.Now()
```

Comment:
in case where there is neither useful nor delayed message we won't use `firstIncludedMsgTime` at all, right? I'm just wondering if there's a chance we would actually use `time.Now()` as some meaningful timestamp

(if we are paranoid then we could make it nillable (through pointer) and then, before final usage in `PostTransaction` assert that it has some value)

Reply:
If I understand correctly, it can still reach PostTransaction with firstIncludedMsgTime == time.Now(), at least when `forcePostBatch := config.MaxDelay <= 0`

Reply:
@Tristan-Wilson isn't that a problem?